recipes/full_finetune_single_device.py

```diff
@@ -569,7 +613,8 @@ def _loss_step(self, batch: Dict[str, torch.Tensor]) -> torch.Tensor:
         # Shape [b, s], needed for the loss not the model
         labels = batch.pop("labels")
 
-        logits = self._model(**batch)
+        with self.activations_handling_ctx:
```

Comment:
Curiosity brings me here. If I gather correctly, every model forward pass for which which we want to offload activations for needs to sit inside this context manager?

Reply:
correct!